sdk/python/kfp/dsl/legacy_data_passing_adaptor.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2021 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Adaptors to support legacy data passing.
+
+The adaptor offers backward compatiblity for pipelines utilize legacy data
+passing. Passing parameters to artifacts or vice versa could result unexpected
+runtime errors. It is highly recommended to not rely on the adaptors. We may
```

Comment:
> Should this file be under compiler package?

Given this is mostly used in `dsl._component_bridge`, I think `dsl` package would be better. Otherwise, the `dsl` package would have a dependency on the `compiler` package.

Reply:
>Should this file be under compiler package?

Architecturally, the `_component_bridge` module should be a part of the compiler. The reason it's in `/dsl/` is that the bridge is installed in the `Pipeline` context class (unrelated to the `@pipeline`), which should also be in `/compiler/`, but is currently in `/dsl/`.
It would be great to do a refactoring to move `Pipeline` and `_component_bridge` to `/compiler/`.

This also simplifies the code for the different compilers, because each compiler can have its own component bridge.

Until this refactoring is done, this code has to remain in `/dsl/` due to the dependency and isolation that Chen mentioned:

>Given this is mostly used in dsl._component_bridge, I think dsl package would be better. Otherwise, the dsl package would have a dependency on the compiler package.